src/layer/S2L1CLayer.ts

```diff
@@ -28,4 +28,14 @@ export class S2L1CLayer extends AbstractSentinelHubV3Layer {
       hasMore: response.data.hasMore,
     };
   }
+
+  public async findDates(
+    bbox: BBox,
+    fromTime: Date,
+    toTime: Date,
+    maxCloudCoverage?: number,
+  ): Promise<Date[]> {
+    const response = await this.fetchDates(bbox, fromTime, toTime, maxCloudCoverage);
+    return response.data.map(date => new Date(date));
+  }
```

Comment:
Maybe instead of defining `findDates()` in every subclass, we could simply rename `fetchDates()` to `findDates()` (on the parent) and adapt it so that it returns an array of `Date` objects?

Reply:
I agree, will change.

Reply:
Changed. Can you please look if this approach is ok?

Reply:
`findDates()` for all `Layer`s is now done similar to how `findTiles` is done for EOCloud layers.